src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -11,27 +12,60 @@ std::optional<Point> CreaseDefenderFSM::findBlockThreatPoint(
     const TbotsProto::CreaseDefenderAlignment& crease_defender_alignment,
     double robot_obstacle_inflation_factor)
 {
+
+    // Look directly at the point
+    // Return the segments that form the path around the crease that the
+    // defenders must follow. It's basically the crease inflated by one robot radius
+    // multiplied by a factor
+    double robot_radius_expansion_amount =
+            ROBOT_MAX_RADIUS_METERS * robot_obstacle_inflation_factor;
+    Rectangle inflated_defense_area =
+            field.friendlyDefenseArea().expand(robot_radius_expansion_amount);
+
     // We increment the angle to positive goalpost by 1/6, 3/6, or 5/6 of the shot
     // cone
-    Angle shot_angle_sixth = convexAngle(field.friendlyGoalpostPos(), enemy_threat_origin,
-                                         field.friendlyGoalpostNeg()) /
-                             6.0;
-    Angle angle_to_positive_goalpost =
-        (field.friendlyGoalpostPos() - enemy_threat_origin).orientation();
-    Angle angle_to_block = angle_to_positive_goalpost + shot_angle_sixth * 3.0;
-    if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::LEFT)
-    {
-        angle_to_block = angle_to_positive_goalpost + shot_angle_sixth * 1.0;
-    }
-    else if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::RIGHT)
-    {
-        angle_to_block = angle_to_positive_goalpost + shot_angle_sixth * 5.0;
-    }
 
+    Vector block_vec = (field.friendlyGoalCenter() - enemy_threat_origin);
+    Angle angle_to_block = block_vec.orientation();
     // Shot ray to block
     Ray ray(enemy_threat_origin, angle_to_block);
+    std::optional<Point> block_point = findDefenseAreaIntersection(field, ray, inflated_defense_area);
+    double x_shift_offset = crease_defender_config.defender_shift_offset_x();
+    double y_shift_offset = crease_defender_config.defender_shift_offset_y();
+    if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::LEFT && block_point.has_value())
+    {
+        // rotate pos 90 deg
+        block_point = stepAlongPerimeter(inflated_defense_area,block_point.value(),-ROBOT_MAX_RADIUS_METERS*2);
+        // Out of field bounds, adjust
+        if (!contains(field.fieldLines(),block_point.value())){
+            /*  In this scenario 0 moves towards x to come back in play.
```

Comment:
i assume the defense area has a bottom, i.e the block point cannot move more than 1 robot radius out of bounds. cuz otherwise the below moving logic wouldn't work right? maybe add a comment to clarify this

Reply:
fixed